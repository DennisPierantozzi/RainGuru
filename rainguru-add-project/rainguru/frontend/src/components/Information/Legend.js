import React, { Component } from "react";

export default class Legend extends Component {
    constructor(props) {
        super(props);
    }

    /**
     * Renders the legend.
     * It renders the following:
     *  - Render the top text above the legend.
     *  - Render the legend with some numbers following a logarithmic scale with rain intensity.
     *
     * @returns the rendered legend.
     */
    render() {
        return (
            <div className="legendContents" data-testid="legend-box">
                <div className="legendTitle">Precipitation [mm/h]</div>
                <div className="legendContainer">
                    <div className="legend">
                        <div id="gradient1container" className="gradient-container">
                            <div id="gradient1" className="gradient-part"></div>
                            <div className="legend-value"><p>0.1</p><i>light</i></div>
                        </div>
                        <div id="gradient2container" className="gradient-container">
                            <div id="gradient2" className="gradient-part"></div>
                        </div>
                        <div id="gradient3container" className="gradient-container">
                            <div id="gradient3" className="gradient-part"></div>
                            <div className="legend-value"><p>1</p><i className="innerRainDescription">moderate</i></div>
                        </div>
                        <div id="gradient4container" className="gradient-container">
                            <div id="gradient4" className="gradient-part"></div>
                            <div className="double-text-div">
                                <div id="left-value" className="legend-value"><p>10</p><i className="innerRainDescription">heavy</i></div>
                                <div id="right-value" className="legend-value"><p>&gt;100</p><i>extreme</i></div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        );
    }


}