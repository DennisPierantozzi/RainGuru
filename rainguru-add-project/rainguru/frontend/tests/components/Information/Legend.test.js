import React from "react";
import {render, screen} from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";
import Legend from "../../../src/components/Information/Legend";

describe("Legend tests", () => {

    it("matches snapshot", () => {
        const state = renderer.create(<Legend/>).toJSON();
        expect(state).toMatchSnapshot();
    });


    it("renders legend", () => {
        const {getByTestId} = render(<Legend/>);
        expect(getByTestId("legend-box")).toBeTruthy();
    });

    it("renders legend text correctly", () => {
        const {getByTestId} = render(<Legend/>);
        expect(getByTestId("legend-box")).toHaveTextContent("Precipitation [mm/h]");
        expect(getByTestId("legend-box")).toHaveTextContent("0.1");
        expect(getByTestId("legend-box")).toHaveTextContent("1");
        expect(getByTestId("legend-box")).toHaveTextContent("10");
        expect(getByTestId("legend-box")).toHaveTextContent(">100");
    });
});