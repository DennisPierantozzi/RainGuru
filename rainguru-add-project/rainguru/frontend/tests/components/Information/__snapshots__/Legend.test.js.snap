// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legend tests matches snapshot 1`] = `
<div
  className="legendContents"
  data-testid="legend-box"
>
  <div
    className="legendTitle"
  >
    Precipitation [mm/h]
  </div>
  <div
    className="legendContainer"
  >
    <div
      className="legend"
    >
      <div
        className="gradient-container"
        id="gradient1container"
      >
        <div
          className="gradient-part"
          id="gradient1"
        />
        <div
          className="legend-value"
        >
          <p>
            0.1
          </p>
          <i>
            light
          </i>
        </div>
      </div>
      <div
        className="gradient-container"
        id="gradient2container"
      >
        <div
          className="gradient-part"
          id="gradient2"
        />
      </div>
      <div
        className="gradient-container"
        id="gradient3container"
      >
        <div
          className="gradient-part"
          id="gradient3"
        />
        <div
          className="legend-value"
        >
          <p>
            1
          </p>
          <i
            className="innerRainDescription"
          >
            moderate
          </i>
        </div>
      </div>
      <div
        className="gradient-container"
        id="gradient4container"
      >
        <div
          className="gradient-part"
          id="gradient4"
        />
        <div
          className="double-text-div"
        >
          <div
            className="legend-value"
            id="left-value"
          >
            <p>
              10
            </p>
            <i
              className="innerRainDescription"
            >
              heavy
            </i>
          </div>
          <div
            className="legend-value"
            id="right-value"
          >
            <p>
              &gt;100
            </p>
            <i>
              extreme
            </i>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
